src/daft-core/src/array/ops/utf8.rs

```diff
@@ -450,6 +450,124 @@ impl Utf8Array {
+                    )),
+                    Some(n_scalar_value) => {
+                        let n_scalar_value: usize =
+                            NumCast::from(n_scalar_value).ok_or_else(|| {
+                                DaftError::ComputeError(format!(
+                                    "failed to cast rhs as usize {n_scalar_value}"
+                                ))
+                            })?;
+                        let arrow_result = self_arrow
+                            .iter()
+                            .map(|val| {
+                                let v = val?;
+                                let len = v.chars().count();
+                                if n_scalar_value == 0 || len == 0 {
+                                    Some(String::new())
+                                } else if n_scalar_value >= len {
+                                    Some(v.to_string())
+                                } else {
+                                    Some(v.chars().skip(len - n_scalar_value).collect::<String>())
+                                }
+                            })
+                            .collect::<arrow2::array::Utf8Array<i64>>();
+
+                        Ok(Utf8Array::from((self.name(), Box::new(arrow_result))))
+                    }
+                }
+            }
+            // broadcast self
+            (1, n_len) => {
+                let self_scalar_value = self.get(0);
+                match self_scalar_value {
+                    None => Ok(Utf8Array::full_null(self.name(), self.data_type(), n_len)),
+                    Some(self_scalar_value) => {
+                        let len = self_scalar_value.chars().count();
+                        let arrow_result = n_arrow
+                            .iter()
+                            .map(|n| match n {
+                                None => Ok(None),
+                                Some(n) => {
+                                    let n: usize = NumCast::from(*n).ok_or_else(|| {
+                                        DaftError::ComputeError(format!(
+                                            "failed to cast rhs as usize {n}"
+                                        ))
+                                    })?;
+                                    if len == 0 || n == 0 {
+                                        Ok(Some(String::new()))
+                                    } else if n >= len {
+                                        Ok(Some(self_scalar_value.to_string()))
+                                    } else {
+                                        Ok(Some(
+                                            self_scalar_value
+                                                .chars()
+                                                .skip(len - n)
+                                                .collect::<String>(),
+                                        ))
+                                    }
```

Comment:
Nit: Seems like each match arm has this duplicate logic for computing the right most nchars of a string, could you extract this out into a separate function?